commit d57bcc3faa4f89efda6a3622dec7182c68b508c1
Author: Zhou Shaoming <zsm@dw.gdbbk.com>
Date:   Fri Oct 27 11:24:14 2017 +0800

    change natural orientation to portrait

diff --git a/device/bbkedu/OK100/ProjectConfig.mk b/device/bbkedu/OK100/ProjectConfig.mk
index 3ed2fb6..3e0dbf2 100755
--- a/device/bbkedu/OK100/ProjectConfig.mk
+++ b/device/bbkedu/OK100/ProjectConfig.mk
@@ -445,7 +445,7 @@ MTK_K64_SUPPORT = yes
 MTK_KERNEL_POWER_OFF_CHARGING = yes
 MTK_LAUNCHER_UNREAD_SUPPORT = yes
 MTK_LCM_DEVICE_TREE_SUPPORT = no
-MTK_LCM_PHYSICAL_ROTATION = 270
+MTK_LCM_PHYSICAL_ROTATION = 0
 MTK_LIVEWALLPAPER_APP = no
 MTK_LOG2SERVER_APP = no
 MTK_LOG2SERVER_INTERNAL = no
diff --git a/frameworks/base/core/res/res/values/config.xml b/frameworks/base/core/res/res/values/config.xml
index cf8b584..f053bfe 100755
--- a/frameworks/base/core/res/res/values/config.xml
+++ b/frameworks/base/core/res/res/values/config.xml
@@ -638,7 +638,7 @@
          landscape will cause a clockwise rotation, and on a landscape device an
          app requesting portrait will cause a counter-clockwise rotation.  Setting
          true here reverses that logic. -->
-    <bool name="config_reverseDefaultRotation">false</bool>
+    <bool name="config_reverseDefaultRotation">true</bool>
 
     <!-- Sets the minimum and maximum tilt tolerance for each possible rotation.
          This array consists of 4 pairs of values which specify the minimum and maximum
diff --git a/frameworks/base/services/core/java/com/android/server/policy/PhoneWindowManager.java b/frameworks/base/services/core/java/com/android/server/policy/PhoneWindowManager.java
index b18fbc9..3819c12 100755
--- a/frameworks/base/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/frameworks/base/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -1976,10 +1976,8 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 mPortraitRotation = Surface.ROTATION_90;
                 mUpsideDownRotation = Surface.ROTATION_270;
             } else {
-		//eebbk <BSP> <liudj> <20171011>  270 and 90 switch begin
-                mPortraitRotation = Surface.ROTATION_90;
-                mUpsideDownRotation = Surface.ROTATION_270;
-		//eebbk <BSP> <liudj> <20171011> modify 270 and 90 switch end
+                mPortraitRotation = Surface.ROTATION_270;
+                mUpsideDownRotation = Surface.ROTATION_90;
             }
         } else {
             shortSize = width;
diff --git a/kernel-4.4/arch/arm64/boot/dts/mediatek/OK100.dts b/kernel-4.4/arch/arm64/boot/dts/mediatek/OK100.dts
index 5c5d9cc..5b9daa4 100755
--- a/kernel-4.4/arch/arm64/boot/dts/mediatek/OK100.dts
+++ b/kernel-4.4/arch/arm64/boot/dts/mediatek/OK100.dts
@@ -32,7 +32,7 @@
 		compatible = "mediatek,lis3dh";
 		i2c_num	= <1>;
 		i2c_addr = <0x19 0 0 0>;
-		direction = <5>;  //6
+		direction = <4>;  //6
 		power_id = <0xffff>;
 		power_vol = <0>;
 		firlen = <0>;
diff --git a/kernel-4.4/arch/arm64/configs/OK100_debug_defconfig b/kernel-4.4/arch/arm64/configs/OK100_debug_defconfig
index 940488e..359f0ee 100755
--- a/kernel-4.4/arch/arm64/configs/OK100_debug_defconfig
+++ b/kernel-4.4/arch/arm64/configs/OK100_debug_defconfig
@@ -199,7 +199,7 @@ CONFIG_MTK_SYNC=y
 CONFIG_MTK_VIDEOCODEC_DRIVER=y
 CONFIG_MTK_FB=y
 CONFIG_MTK_VIDEOX=y
-CONFIG_MTK_LCM_PHYSICAL_ROTATION="270"
+CONFIG_MTK_LCM_PHYSICAL_ROTATION="0"
 CONFIG_LCM_HEIGHT="1280"
 CONFIG_LCM_WIDTH="800"
 CONFIG_MTK_SENSOR_SUPPORT=y
diff --git a/kernel-4.4/arch/arm64/configs/OK100_defconfig b/kernel-4.4/arch/arm64/configs/OK100_defconfig
index 3e5dd7c..9489aef 100755
--- a/kernel-4.4/arch/arm64/configs/OK100_defconfig
+++ b/kernel-4.4/arch/arm64/configs/OK100_defconfig
@@ -194,7 +194,7 @@ CONFIG_MTK_SYNC=y
 CONFIG_MTK_VIDEOCODEC_DRIVER=y
 CONFIG_MTK_FB=y
 CONFIG_MTK_VIDEOX=y
-CONFIG_MTK_LCM_PHYSICAL_ROTATION="270"
+CONFIG_MTK_LCM_PHYSICAL_ROTATION="0"
 CONFIG_LCM_HEIGHT="1280"
 CONFIG_LCM_WIDTH="800"
 CONFIG_MTK_SENSOR_SUPPORT=y
diff --git a/kernel-4.4/drivers/input/touchscreen/mediatek/focaltech_touch/focaltech_core.c b/kernel-4.4/drivers/input/touchscreen/mediatek/focaltech_touch/focaltech_core.c
index de0a2cc..99649d4 100755
--- a/kernel-4.4/drivers/input/touchscreen/mediatek/focaltech_touch/focaltech_core.c
+++ b/kernel-4.4/drivers/input/touchscreen/mediatek/focaltech_touch/focaltech_core.c
@@ -380,10 +380,8 @@ static void tpd_down(int x, int y, int p, int id)
 #endif
 
     input_report_abs(tpd->dev, ABS_MT_TOUCH_MAJOR, 1);
-    //input_report_abs(tpd->dev, ABS_MT_POSITION_X, x);
-    //input_report_abs(tpd->dev, ABS_MT_POSITION_Y, y);
-    input_report_abs(tpd->dev, ABS_MT_POSITION_X, 1279-y);
-    input_report_abs(tpd->dev, ABS_MT_POSITION_Y, x);
+    input_report_abs(tpd->dev, ABS_MT_POSITION_X, x);
+    input_report_abs(tpd->dev, ABS_MT_POSITION_Y, y);
     input_mt_sync(tpd->dev);
 
     FTS_DEBUG("[A]P%d(%4d,%4d)[p:%d] DOWN!", id, x, y, p);
@@ -704,10 +702,8 @@ static int fts_report_touch(struct ts_event *data)
             }
             input_report_abs(tpd->dev, ABS_MT_TOUCH_MAJOR, data->area[i]);
 
-            //input_report_abs(tpd->dev, ABS_MT_POSITION_X,data->au16_x[i]);
-            //input_report_abs(tpd->dev, ABS_MT_POSITION_Y,data->au16_y[i]);
-            input_report_abs(tpd->dev, ABS_MT_POSITION_X,1279-data->au16_y[i]);
-            input_report_abs(tpd->dev, ABS_MT_POSITION_Y,data->au16_x[i]);
+            input_report_abs(tpd->dev, ABS_MT_POSITION_X,data->au16_x[i]);
+            input_report_abs(tpd->dev, ABS_MT_POSITION_Y,data->au16_y[i]);
             touchs |= BIT(data->au8_finger_id[i]);
             data->touchs |= BIT(data->au8_finger_id[i]);
 
diff --git a/vendor/mediatek/proprietary/bootable/bootloader/lk/project/OK100.mk b/vendor/mediatek/proprietary/bootable/bootloader/lk/project/OK100.mk
index 93a2872..4d70998 100644
--- a/vendor/mediatek/proprietary/bootable/bootloader/lk/project/OK100.mk
+++ b/vendor/mediatek/proprietary/bootable/bootloader/lk/project/OK100.mk
@@ -11,7 +11,7 @@ DEFINES += MTK_NEW_COMBO_EMMC_SUPPORT
 MTK_KERNEL_POWER_OFF_CHARGING = yes
 #DEFINES += SWCHR_POWER_PATH
 #DEFINES += MTK_BQ24261_SUPPORT
-MTK_LCM_PHYSICAL_ROTATION = 270
+MTK_LCM_PHYSICAL_ROTATION = 0
 CUSTOM_LK_LCM="kd070d5450nha6_rgb_dpi hx83100_wxga_dsi_vdo_innolux"
 #nt35595_fhd_dsi_cmd_truly_nt50358 = yes
 
diff --git a/vendor/mediatek/proprietary/custom/mt8167/hal/imgsensor_metadata/common/config_static_metadata_common.h b/vendor/mediatek/proprietary/custom/mt8167/hal/imgsensor_metadata/common/config_static_metadata_common.h
index 80f7209..d4c8f18 100644
--- a/vendor/mediatek/proprietary/custom/mt8167/hal/imgsensor_metadata/common/config_static_metadata_common.h
+++ b/vendor/mediatek/proprietary/custom/mt8167/hal/imgsensor_metadata/common/config_static_metadata_common.h
@@ -62,11 +62,11 @@ STATIC_METADATA_BEGIN(DEVICE, CAMERA, COMMON)
     case 0:
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_ORIENTATION)
-            CONFIG_ENTRY_VALUE(180, MINT32)
+            CONFIG_ENTRY_VALUE(90, MINT32)
         CONFIG_METADATA_END()
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_WANTED_ORIENTATION)
-            CONFIG_ENTRY_VALUE(180, MINT32)
+            CONFIG_ENTRY_VALUE(90, MINT32)
         CONFIG_METADATA_END()
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_FACING)
@@ -85,11 +85,11 @@ STATIC_METADATA_BEGIN(DEVICE, CAMERA, COMMON)
     case 1:
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_ORIENTATION)
-            CONFIG_ENTRY_VALUE(180, MINT32)
+            CONFIG_ENTRY_VALUE(270, MINT32)
         CONFIG_METADATA_END()
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_WANTED_ORIENTATION)
-            CONFIG_ENTRY_VALUE(180, MINT32)
+            CONFIG_ENTRY_VALUE(270, MINT32)
         CONFIG_METADATA_END()
         //======================================================================
         CONFIG_METADATA_BEGIN(MTK_SENSOR_INFO_FACING)
